src/Tribe/Extension.php

```diff
@@ -0,0 +1,493 @@
+<?php
+defined( 'WPINC' ) or die; // Do not load directly.
+
+/**
+ * Base Extension class
+ *
+ * Avoid using static states within this class unless it's global for all extensions.
+ * Some extension authors may lack a thorough understanding of OOP and inheritance.
+ * This is built with such ones in mind.
+ *
+ * @package Tribe
+ * @subpackage Extension
+ * @since 4.3.1
+ */
+class Tribe__Extension {
+
+	/**
+	 * Extension arguments
+	 *
+	 * @var array {
+	 *      Each argument for this extension instance
+	 *
+	 *      @type string $version Extension's semantic version number.
+	 *                            Can be manually set in child to boost performance.
+	 *
+	 *      @type string $url     Extension's tec.com page.
+	 *
+	 *      @type string $hook    Action/hook to fire init() on.
+	 *
+	 *      @type string $file    File path containing plugin header.
+	 *                            Can be manually set in child to boost performance.
+	 *
+	 *      @type string $class   Extension's main class name.
+	 *
+	 *      @type array  $requires {
+	 *          Each plugin this extension requires
+	 *
+	 *          @type string $main_class Minimum version number
+	 *      }
+	 *
+	 *      @type array  $meta_links {
+	 *          Meta links added to extension in WP Admin > Plugins page
+	 *
+	 *          @type string $name URL
+	 *      }
+	 *
+	 *      @type array $plugin_data If the plugin file header is parsed, the
+	 *                               resulting data is stored in this.
+	 * }
+	 */
+	protected $_args = array();
+
+	/**
+	 * The various extension instances
+	 *
+	 * If you directly modify this inside of a child class then you're arguably
+	 * an evil person.
```

Comment:
Ah ok. It wouldn't actually override the visibility in that case, though - it would define a new property of protected scope. The contents of the private array here in the base class would remain untouched.